src/UXClient/Models/LineChartData.ts

```diff
@@ -3,18 +3,21 @@ import {ChartComponentData} from "./ChartComponentData";
 
 class LineChartData extends ChartComponentData {
     public timeMap: any = {};
-    get yExtents(): {}{
+    get yExtents(): Array<any>{
         return this._yExtents;
     };
 
-    private _yExtents: {} = {};
+    private _yExtents: Array<any> = [];
     
-    public setYExtents(key: number, value: [number, number]){
-        this._yExtents[key] = value;
+    public setYExtents(idx: number, value: [number, number]){
+        this._yExtents[idx] = value;
     }
 
-    public resetYExtents(){
-        this._yExtents = {};
+    public resetYExtents(numAggs){
```

Comment:
does this need to be passed in?  I would guess LineChartData knows about the length of chartData.  moreover, chartDataOptions are optional, so you should actually use the length of chartData?